ibm_cloud_sdk_core/base_service.py

```diff
@@ -45,19 +46,21 @@ class BaseService(object):
     ICP_PREFIX = 'icp-'
     APIKEY = 'apikey'
     IAM_ACCESS_TOKEN = 'iam_access_token'
+    ICP4D_ACCESS_TOKEN = 'icp4d_access_token'
     URL = 'url'
     USERNAME = 'username'
     PASSWORD = 'password'
     IAM_APIKEY = 'iam_apikey'
     IAM_URL = 'iam_url'
+    ICP4D_URL = 'icp4d_url'
```

Comment:
The url for token exchange looks different from the endpoint for API calls. This was based on the test instance provided:

```
assistant = AssistantV1(
    version='2018-07-10',
    username='admin',
    password='',
    url='https://mycluster.icp:31843/assistant/cfg-default/instances/1559584393/api',
    icp4d_url='https://mycluster.icp:31843',
    authentication_type='icp4d')
assistant.disable_SSL_verification()
```

Reply:
Oh, interesting. I hadn't seen that yet. Still, I wonder if there is a concrete way to parse the base URL from the service URL to use in the token manager. It seems redundant to pass in both of those when the base is the same. What do you think?

Reply:
My 2 cents...  while it might be the case that the service endpoint and icp4d token service share a common part of their respective URLs, what happens when the architecture or configuration changes and that is no longer the case?   It's probably best to let the user specify the service endpoint URL separate from the ICP4D token service url.

Reply:
I too agree with @padamstx to have a separate url for the two.

Reply:
Okay, after talking with Phil a bit, I am convinced on accepting two separate URLs. That said, how we specifically handle that might be worth some additional discussion. Let's follow up on this tomorrow

Reply:
Let's discuss this on the 6/5 standup call and make sure all SDK cores (and generator) are on the same page with this.